lib/Dialect/Torch/Transforms/FuseQuantizedOps.cpp

```diff
@@ -72,51 +76,61 @@ class QuantizeOperandsPastCommutingOps : public OpRewritePattern<SrcOp> {
       return rewriter.notifyMatchFailure(
           op, "1-d quantized convolution is not supported");
 
+    SmallVector<QuantizedChain, 2> operandChains;
     for (unsigned i : QuantInfo<SrcOp>::operandsToQuantize) {
       Value operand = operands[i];
-      std::stack<mlir::Operation *> commutingOpStack;
-      Value dequantOpd, MPTQTOpd, scale, zeroPoint;
+      QuantizedChain chain;
       for (unsigned k = 0; k < depth + 1; k++) {
         auto currOp = operand.getDefiningOp();
         // Case 0 : currOp is a nullptr (e.g., operand is a block argument)
         if (!currOp)
           break;
         // Case 1 : currOp is a q commuting op (continue loop)
         if (isQCommutingOp(currOp)) {
-          commutingOpStack.push(currOp);
+          chain.commutingOpStack.push(currOp);
           // set operand to currOp for next k-iteration
           operand = currOp->getOperand(0);
           continue;
         }
         // Case 2 : currOp is a dequant op (end loop)
         if (llvm::isa<AtenDequantizeSelfOp, AtenDequantizeTensorOp>(currOp)) {
-          dequantOpd = currOp->getOperand(0);
+          chain.dequantOpd = currOp->getOperand(0);
+          // Bail out if any operand is per-channel quantized, which would
+          // require more complex fusion logic.
+          if (llvm::isa<Aten_MakePerChannelQuantizedTensorOp>(
+                  chain.dequantOpd.getDefiningOp()))
+            return rewriter.notifyMatchFailure(
+                op, "fusion with per-channel quantization is not supported");
+
           auto MPTQTOp =
-              dequantOpd.getDefiningOp<Aten_MakePerTensorQuantizedTensorOp>();
-          MPTQTOpd = MPTQTOp.getOperand(0);
-          scale = MPTQTOp.getOperand(1);
-          zeroPoint = MPTQTOp.getOperand(2);
+              chain.dequantOpd
+                  .getDefiningOp<Aten_MakePerTensorQuantizedTensorOp>();
+          chain.MPTQTOpd = MPTQTOp.getOperand(0);
+          chain.scale = MPTQTOp.getOperand(1);
+          chain.zeroPoint = MPTQTOp.getOperand(2);
         }
         // either a dequant was found or chain broken, so break loop
         break;
       }
 
       // move to next operand if this trace was unsuccessful
-      if (!MPTQTOpd)
+      if (!chain.MPTQTOpd)
         continue;
 
       // a successful trace occured, so set dequant to true
-      dequanted = true;
+      operandChains.push_back(std::move(chain));
+    }
```

Comment:
Probably don't need the early continue now that there is only one line of code aftewards. I'd like it if you either update the comment about the absent `dequant` bool (if you'd like to keep it like this), or make a change like:
```cpp
  // if trace was successful, add it to operandChains
  if (chain.MPTQTOpd)
    operandChains.push_back(std:move(chain));
}
```